lib/src/layer.dart

```diff
@@ -1,53 +1,54 @@
-part of tiled;
+import 'package:xml/xml.dart';
+import '../tiled.dart';
+import 'tile_map_parser.dart';
+import 'tile_map.dart';
+import 'tile.dart';
+import 'node_dsl.dart';
+import 'flips.dart';
 
 class Layer {
   static const int FLIPPED_HORIZONTALLY_FLAG = 0x80000000;
   static const int FLIPPED_VERTICALLY_FLAG = 0x40000000;
   static const int FLIPPED_DIAGONALLY_FLAG = 0x20000000;
 
-  String name;
-  int width;
-  int height;
-  bool visible;
+  String name = "";
+  int width = 0;
+  int height = 0;
+  bool visible = true;
```

Comment:
Why do these need initialization here when they also come in as non-nullable arguments in the constructor?

Reply:
There are two more constructors (onne named constructor ), and then code would be double, I think this is very clear to read.